script.js

```diff
@@ -1,4 +1,128 @@
+var player1Score = [];
+var player2Score = [];
+
+//add up total scores
+var player1ScoreSum = 0;
+var player2ScoreSum = 0;
+
+//num of wins
+var player1Wins = 0;
+var player2Wins = 0;
+
+//default game mode start w player 1
+currentGameMode = `player 1 roll dice`;
+
+//keep track of player's numbers
+var player1Num;
+var player2Num;
+
+//generate random number 1-6
+var rollDice = function () {
+  var playerRollDice = Math.floor(Math.random() * 6) + 1;
+  return playerRollDice.toString();
+};
+
+var beatThat = function (input) {
+  //start with player 1 rolling dice
+  if (currentGameMode == `player 1 roll dice`) {
+    //add rolled dice to player 1 array
+    var dice1 = rollDice();
+    var dice2 = rollDice();
+    player1List[0] = dice1;
+    player1List[1] = dice2;
+    console.log(player1List);
+
+    currentGameMode = `player 1 choose dice order`;
+    return `Welcome Player 1. <br><br> You rolled ${dice1} for Dice 1 and ${dice2} for Dice 2. <br><br> Please input 1 or 2 to choose the order of the dice`;
+
+    //switch to player 1 choosing dice order to make combined number
+  } else if (currentGameMode == `player 1 choose dice order`) {
+    if (input == 1) {
+      player1Num = player1List[0].toString() + player1List[1].toString();
+      player1Score.push(player1Num);
+    } else if (input == 2) {
+      player1Num = player1List[1].toString() + player1List[0].toString();
+      player1Score.push(player1Num);
+    }
+    player1ScoreSum += Number(player1Num);
+    console.log(`player 1 number is ${player1Num}`);
+
+    currentGameMode = `player 2 roll dice`;
+    return `Player 1 chose Dice ${input} first. <br><br> Your number is ${player1Num}. <br><br> Player 2, please click submit to roll the dice.`;
+
+    //switch to player 2 rolling dice mode
+  } else if (currentGameMode == `player 2 roll dice`) {
+    //add 2 rolled dice to player 2 array
+    var dice1 = rollDice();
+    var dice2 = rollDice();
+    player2List[0] = dice1;
+    player2List[1] = dice2;
+    console.log(player2List);
+
+    currentGameMode = "player 2 choose dice order";
+    return `Welcome Player 2. <br><br> You rolled ${player2List[0]} for Dice 1 and ${player2List[1]} for Dice 2. <br><br> Please input 1 or 2 to choose the order of the dice`;
+
+    //swtich to player 2 choosing dice order for combined number
+  } else if (currentGameMode == `player 2 choose dice order`) {
+    if (input == 1) {
+      var player2Num = player2List[0].toString() + player2List[1].toString();
+      player2Score.push(player2Num);
+    } else if (input == 2) {
+      var player2Num = player2List[1].toString() + player2List[0].toString();
+      player2Score.push(player2Num);
+    }
+
+    player2ScoreSum += Number(player2Num);
+    console.log(`player 2 number is ${player2Num}`);
+    currentGameMode = `compare numbers`;
+    return `Player 2 chose Dice ${input} first. <br><br> Your number is ${player2Num}. <br><br> Click submit to see who wins`;
```

Comment:
This block of code is repeated from above, you might want to refactor it to a more general helper function that can be used for both player 1 and 2! You can use arguments to control where the values should be stored (in `player1List` or `player2List` etc.)